sstables/sstable_set.cc

```diff
@@ -40,8 +40,7 @@ bool sstable_run::will_introduce_overlapping(const shared_sstable& sst) const {
         auto less_cmp = [s = s1->get_schema()] (const dht::decorated_key& k1, const dht::decorated_key& k2) {
             return k1.less_compare(*s, k2);
         };
-        return less_cmp(s1->get_first_decorated_key(), s2->get_first_decorated_key()) &&
-               less_cmp(s1->get_last_decorated_key(), s2->get_first_decorated_key());
+        return less_cmp(s1->get_last_decorated_key(), s2->get_first_decorated_key());
     };
     // lower bound will be the 1st element which is not *all* before the candidate sstable.
```

Comment:
Let's say we have a range tombstone at timestamp t1 and range [x, y] and a row at timestamp t2 > t1 and position x. Will the rules change the range to (x, y]? I think they should, but let's verify. @denesb @tgrabiec

Reply:
Wait this is the partition key, so doesn't apply.

Reply:
Okay, after this next patch we compare clustering keys too, so the question does apply.

Reply:
The resulting range will be `[x, y]`.

Reply:
We don't deoverlap?

Reply:
I don't think we deoverlap in the sstable writer level.

Reply:
But the writer is fed from a flat_mutation_reader_v2, won't it deoverlap?

Reply:
It deoverlaps range tombstones (they are intrinsically deoverlapped) but doesn't split/deoverlap around rows.